Mathlib/Algebra/Module/Equiv/Basic.lean

```diff
@@ -494,24 +529,15 @@ def smulOfUnit (a : Rˣ) : M ≃ₗ[R] M :=
   DistribMulAction.toLinearEquiv R M a
 
 /-- A linear isomorphism between the domains and codomains of two spaces of linear maps gives a
-linear isomorphism between the two function spaces. -/
+linear isomorphism between the two function spaces.
+
+See `LinearEquiv.arrowCongrAddEquiv` for the additive version of this isomorphism that works
+over a not necessarily commutative semiring. -/
```

Comment:
Could we generalize to `(M₁ →ₗ[R] M₂₁) ≃ₗ[S] (M₂ →ₗ[R] M₂₂)` too? That would then let us use `S := MulOpposite R`, as well as `S := Nat` for the additive case.

Reply:
`e₂` needs to be `S`-linear too, so I can't replace the original def:
```
def arrowCongr [Semiring S] [Module S M₂₁] [Module S M₂₂] [SMulCommClass R S M₂₁]
    [SMulCommClass R S M₂₂] (e₁ : M₁ ≃ₗ[R] M₂) (e₂ : M₂₁ ≃ₗ[R] M₂₂) (h : IsLinearMap S e₂) :
    (M₁ →ₗ[R] M₂₁) ≃ₗ[S] M₂ →ₗ[R] M₂₂ where
  __ := arrowCongrAddEquiv e₁ e₂
  map_smul' c f := by ext; simp [arrowCongrAddEquiv, h.map_smul]
```